rpc/frontend/src/main/scala/io/udash/rpc/internals/UsesServerRPC.scala

```diff
@@ -44,24 +46,29 @@ private[rpc] trait UsesServerRPC[ServerRPCType] extends UsesRemoteRPC[ServerRPCT
   }
 
   /** Registers callback which will be called whenever RPC request returns failure. */
-  def registerCallFailureCallback(callback: Throwable => Unit): Registration = {
+  def registerCallFailureCallback(callback: RPCFailureListener): Registration = {
     exceptionCallbacks += callback
     new Registration {
       override def cancel(): Unit = exceptionCallbacks -= callback
     }
   }
 
+  private def handleException(ex: Throwable): Unit =
+    exceptionCallbacks.foreach { pf =>
```

Comment:
This is `.collect`.

Reply:
This is collection of PFs and this `.foreach` applies all of them to provided exception. `.collect` applies provided PF to elements from collection.

Anyway I'll refactor callbacks handing, because it's reusable piece of code.